kork-core/src/main/java/com/netflix/spinnaker/kork/spring/PluginLoader.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.kork.spring;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+public class PluginLoader {
+
+  static final String DEFAULT_PLUGIN_CONFIG_PATH = "/opt/spinnaker/config/plugins.yml";
+
+  private String filename;
+  private List<URL> urls;
+
+  PluginLoader(String filename) {
+    this.filename = filename;
+  }
+
+  PluginLoader() {
+    this.filename =
+        Optional.ofNullable(System.getenv("PLUGIN_CONFIG_LOCATION"))
+            .orElse(PluginLoader.DEFAULT_PLUGIN_CONFIG_PATH);
+  }
+
+  public void loadPlugins(Class source) throws PluginLoaderException {
+    try {
+      InputStream inputStream = openFromFile();
+      PluginProperties props = parsePluginConfigs(inputStream);
+      addPlugins(props);
+    } catch (IOException e) {
+      throw new PluginLoaderException(e);
+    }
+    addJarsToClassPath(source);
+  }
+
+  private PluginProperties parsePluginConfigs(InputStream inputStream)
+      throws PluginLoaderException {
+    ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+    try {
+      PluginProperties pluginConfigs = objectMapper.readValue(inputStream, PluginProperties.class);
+      return pluginConfigs;
+    } catch (IOException e) {
+      throw new PluginLoaderException("Not able to parse plugin configurations: " + e.getMessage());
+    }
+  }
+
+  private InputStream openFromFile() throws IOException {
+    return new FileInputStream(this.filename);
+  }
+
+  public void addPlugins(PluginProperties pluginProperties) throws PluginLoaderException {
+    List<Path> jarPaths = getEnabledJarPaths(pluginProperties);
+    this.urls = jarPathsToUrls(jarPaths);
+  }
+
+  private List<Path> getEnabledJarPaths(PluginProperties pluginProperties)
+      throws PluginLoaderException {
+    try {
+      return pluginProperties.pluginConfigurationList.stream()
+          .filter(isEnabled())
+          .map(PluginProperties.PluginConfiguration::getPath)
+          .map(Paths::get)
+          .collect(Collectors.toList());
+    } catch (Exception e) {
+      throw new PluginLoaderException("Was not able to get plugin jar paths: " + e.getMessage());
+    }
+  }
+
+  private Predicate<PluginProperties.PluginConfiguration> isEnabled() {
+    return p -> p.enabled == true;
+  }
+
+  private List<URL> jarPathsToUrls(List<Path> jars) throws PluginLoaderException {
+    try {
+      return jars.stream().map(this::getUrlfromPath).collect(Collectors.toList());
+    } catch (RuntimeException e) {
```

Comment:
You typically don't catch `RuntimeException`. What type of Exception do you expect? Catch it higher.